File: frontend/src/types/financeiro.ts
/** TypeScript interfaces for FinanceiroMissao entity matching Pydantic schemas. */

import type { FinanceiroStatus } from './enums';

export interface FinanceiroMissaoResponse {
  id: number;
  missao_id: number;
  custo_estimado: number;
  custo_realizado: number;
  valor_faturado: number;
  status_financeiro: FinanceiroStatus;
  observacoes: string | null;
  fechado_por: number | null;
  fechado_em: string | null;
  created_at: string;
  updated_at: string;
}

export interface FinanceiroMissaoUpdate {
  custo_estimado?: number | null;
  custo_realizado?: number | null;
  valor_faturado?: number | null;
  status_financeiro?: FinanceiroStatus | null;
  observacoes?: string | null;
}

File: app/schemas/financeiro.py
"""Pydantic schemas for FinanceiroMissao entity."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, Field

from app.models.enums import FinanceiroStatus


class FinanceiroMissaoBase(BaseModel):
    custo_estimado: Decimal = Field(default=Decimal("0"), ge=0)
    custo_realizado: Decimal = Field(default=Decimal("0"), ge=0)
    valor_faturado: Decimal = Field(default=Decimal("0"), ge=0)
    status_financeiro: FinanceiroStatus = FinanceiroStatus.PENDENTE
    observacoes: str | None = None


class FinanceiroMissaoUpdate(BaseModel):
    custo_estimado: Decimal | None = Field(None, ge=0)
    custo_realizado: Decimal | None = Field(None, ge=0)
    valor_faturado: Decimal | None = Field(None, ge=0)
    status_financeiro: FinanceiroStatus | None = None
    observacoes: str | None = None


class FinanceiroMissaoResponse(FinanceiroMissaoBase):
    id: int
    missao_id: int
    fechado_por: int | None
    fechado_em: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
